sklearn/compose/_column_transformer.py

```diff
@@ -655,11 +664,17 @@ def _add_prefix_for_feature_names_out(self, transformer_with_feature_names_out):
         feature_names_out : ndarray of shape (n_features,), dtype=str
             Transformed feature names.
         """
-        if self.verbose_feature_names_out:
+        feature_names_out_callable = None
+        if callable(self.verbose_feature_names_out):
+            feature_names_out_callable = self.verbose_feature_names_out
+        elif self.verbose_feature_names_out is True:
+            feature_names_out_callable = _feature_names_out
+
+        if feature_names_out_callable is not None:
             # Prefix the feature names out with the transformers name
             names = list(
                 chain.from_iterable(
-                    (f"{name}__{i}" for i in feature_names_out)
+                    (feature_names_out_callable(name, i) for i in feature_names_out)
```

Comment:
What do you think of allowing a string format to change instead of a callable?

```python
feature_names_out_template = "{transformer_name}${feature_name}"

chain.from_iterable(
    feature_names_out_template.format(
        transformer_name=transformer_name,
        feature_name=feature_name
    ) ...
)

```

BTW I am okay with a callable. I'm wondering if a string format is enough flexibility here.

Reply:
I think this is a great idea to also allow for a string to format. It is a quicker way to do basic stuff, but it does not have the flexibility of a callable. For instance, it is impossible to put the transformer name in upper_case using a string to be formatted. Here is the doc about what is possible with string formatting: https://docs.python.org/fr/3.5/library/string.html#format-string-syntax. It shows limited possibilities.